plugins/org.eclipse.n4js/src/org/eclipse/n4js/scoping/utils/CanLoadFromDescriptionHelper.java

```diff
@@ -243,4 +243,11 @@ private boolean projectContainsURI(IN4JSProject project, URI candidate) {
 		return Optional.ofNullable(description).flatMap(UserdataMapper::readDependenciesFromDescription);
 	}
 
+	/**
+	 * Create a resource in the context of the given resource set.
+	 */
+	public Resource createResource(ResourceSet resourceSet, URI resourceURI) {
```

Comment:
Could you please explain what use cases this method is intended for? Because it is not intended (I guess) that _all_ Resource creation be handled by this method, in the future.

If I understand correctly, it's only intended for creation of resources "during load from index", right? (maybe "during load from index" can be worded/explained better)